### PR TITLE
remove 4 weed dependencies, and switch a function from parsec to megaparsec

### PR DESCRIPTION
**Pull request by cleverca22**
## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by cleverca22** [comment]
dang, its using parsers from serokell-util, which use parsec!

File: infra/src/Pos/Infra/DHT/Model/Types.hs [hunk 1]
```diff
@@ -110,11 +110,11 @@ randomDHTKey = DHTKey . hashAddress <$> liftIO (runSecureRandom genNonce)
 ----------------------------------------------------------------------------
 
 -- | Parser for DHT key.
-dhtKeyParser :: P.Parser DHTKey
+dhtKeyParser :: P.CharParser DHTKey
 dhtKeyParser = P.base64Url >>= toDHTKey
   where
-    toDHTKey = either P.parserFail return . bytesToDHTKey
+    toDHTKey = either undefined return . bytesToDHTKey
```
Comment (cleverca22) [inline infra/src/Pos/Infra/DHT/Model/Types.hs]: todo: dont use `undefined`!!!

File: infra/src/Pos/Infra/DHT/Real/CLI.hs [hunk 2]
```diff
@@ -61,7 +64,11 @@ dhtPeersFileOption =
 readDhtPeersFile :: FilePath -> IO [DHTNode]
 readDhtPeersFile path = do
     xs <- lines <$> readFile path
-    let parseLine x = case parse (dhtNodeParser <* eof) "" x of
+    let
+      ugh :: (Token s ~ Char, Tokens s ~ String, Stream s) => ParsecT () s m DHTNode
```
Comment (cleverca22) [inline infra/src/Pos/Infra/DHT/Real/CLI.hs]: todo: fix this type

File: util/Pos/Util/OptParse.hs [hunk 3]
```diff
@@ -7,6 +7,6 @@ import           Universum
 import           Options.Applicative (ReadM, eitherReader)
 import           Text.Megaparsec (Parsec, parse)
 
-fromParsec :: Show e => Parsec e Text a -> ReadM a
+fromParsec :: Parsec () String a -> ReadM a
```
Comment (cleverca22) [inline util/Pos/Util/OptParse.hs]: and maybe put e back here?

**Comment by jcmincke** [comment]
@cleverca22 Can we close this PR?

**Pull request closed without merging**